File: Ruby/test/test_reverse_words_in_a_string.rb
require './reverse_words_in_a_string.rb'
require 'test/unit'

class TestReverseWordsInAString < Test::Unit::TestCase
  def test_reverse_words_in_a_string
    strings = [
      'the sky is blue',
      ' the sky is blue ',
      ' the sky     is blue '
    ]
    strings.each do |string|
      assert_equal('blue is sky the', reverse_words_in_a_string(string))
    end
  end
end

File: Nodejs/test/test_reverse_words_in_a_string.js
var assert = require('chai').assert;
var reverseWords = require('../reverse_words_in_a_string');

describe('ReverseWordsInAString', function() {
  it('test reverseWords', function() {
	  [
	    'the sky is blue',
	    ' the sky is blue ',
	    ' the sky     is blue '
	  ].forEach(function(string){
	  	assert.equal('blue is sky the', reverseWords(string));
	  });
  });
});